packages/template-retail-react-app/app/hooks/use-current-basket.js

```diff
@@ -4,52 +4,40 @@
  * SPDX-License-Identifier: BSD-3-Clause
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
-import {useCustomerBaskets, useCustomerId, useProducts} from 'commerce-sdk-react-preview'
+import {useCustomerBaskets, useShopperBasketsMutation} from 'commerce-sdk-react-preview'
+import {useCurrentCustomer} from './use-current-customer'
 
+const onClient = typeof window !== 'undefined'
 /**
  * This hook combine some commerce-react-sdk hooks to provide more derived data for Retail App baskets
  * @param id - basket id to get the current used basket among baskets returned, use first basket in the array if not defined
  * @param shouldFetchProductDetail - boolean to indicate if the baskets should fetch product details based on basket items
  */
-export const useCurrentBasket = ({id = '', shouldFetchProductDetail = false} = {}) => {
-    const customerId = useCustomerId() || ''
-    const {
-        data: basketsData,
-        isLoading: isBasketsLoading,
-        error: basketsError
-    } = useCustomerBaskets({parameters: {customerId}})
-    // if id is not defined, by default use the first basket in the list
-    const basket =
-        basketsData?.baskets?.find((basket) => basket.basketId === id) || basketsData?.baskets?.[0]
-    const productIds = basket?.productItems?.map(({productId}) => productId).join(',') ?? ''
-    const {
-        data: products,
-        isLoading: isProductsLoading,
-        error: productsError
-    } = useProducts(
-        {
-            parameters: {
-                ids: productIds,
-                allImages: true
-            }
-        },
+export const useCurrentBasket = ({id = ''} = {}) => {
+    const createBasket = useShopperBasketsMutation('createBasket')
+    const {data: customer} = useCurrentCustomer()
+    const {customerId} = customer
+    const {data: basketsData, ...restOfQuery} = useCustomerBaskets(
+        {parameters: {customerId}},
         {
-            enabled: shouldFetchProductDetail && !!productIds,
-            select: (result) => {
-                // Convert array into key/value object with key is the product id
-                return result?.data?.reduce((result, item) => {
-                    const key = item.id
-                    result[key] = item
-                    return result
-                }, {})
+            enabled: !!customerId && onClient,
+            onSuccess: (data) => {
+                if (!data.total) {
+                    console.log('creating basket')
+                    createBasket.mutate({
+                        body: {}
+                    })
```

Comment:
quick question, does this mutate invalidate `useCustomerBaskets` and triggers a refetch?

Reply:
No, we do not invalidate for this mutation, we update the cache manually when mutation succeeds